contracts/ensWildcardResolvers/DigidaigakuResolver.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "./WildcardResolverBase.sol";
+import "../libraries/StringParsing.sol";
+import "../ensGuilds/GuildsResolver.sol";
+import "@ensdomains/ens-contracts/contracts/reverseRegistrar/ReverseClaimer.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract DigidaigakuResolver is WildcardResolverBase, ReverseClaimer, Ownable {
+    using StringParsing for bytes;
+    using Strings for string;
+    using Strings for address;
+    using Strings for uint256;
+
+    string public url;
+    IERC721 digidaigakuContract;
+
+    constructor(
+        ENS _ensRegistry,
+        INameWrapper _ensNameWrapper,
+        address reverseRecordOwner,
+        address tokenContract,
+        string memory _url
+    ) WildcardResolverBase(_ensRegistry, _ensNameWrapper) ReverseClaimer(_ensRegistry, reverseRecordOwner) {
+        digidaigakuContract = IERC721(tokenContract);
+        url = _url;
+        return;
+    }
+
+    function setUrl(string memory new_url) public onlyOwner {
+        url = new_url;
+    }
+
+    function _resolveWildcardEthAddr(
+        bytes calldata childUtf8Encoded,
+        bytes calldata parentDnsEncoded
+    ) internal view virtual override returns (address) {
+        // Extract tokenId from child name
+        (bool valid, uint256 tokenId) = childUtf8Encoded.parseUint256();
+        // No token, try resolving using name
+        if (!valid) {
+            string[] memory urls = new string[](1);
+            urls[0] = url;
+            bytes memory extraData = abi.encode(parentDnsEncoded);
+
+            revert OffchainLookup(
+                address(this),
+                urls,
+                childUtf8Encoded,
+                this.resolveByNameCallback.selector,
+                extraData
+            );
```

Comment:
Nitpick: do we need to abi.encode the parent name? Could we just do this and save some gas?
```suggestion
            revert OffchainLookup(
                address(this),
                urls,
                childUtf8Encoded,
                this.resolveByNameCallback.selector,
                parentDnsEncoded
            );
```